src/compile/legend/parse.ts

```diff
@@ -14,16 +14,20 @@ import {
 import {FieldDef, isFieldDef, title as fieldDefTitle} from '../../fielddef';
 import {Legend, LEGEND_PROPERTIES, VG_LEGEND_PROPERTIES} from '../../legend';
 import {GEOJSON} from '../../type';
-import {deleteNestedProperty, getFirstDefined, keys} from '../../util';
+import {deleteNestedProperty, getFirstDefined, keys, varName} from '../../util';
 import {guideEncodeEntry, mergeTitleComponent, numberFormat} from '../common';
 import {isUnitModel, Model} from '../model';
 import {parseGuideResolve} from '../resolve';
+import {assembleLegendSelection} from '../selection/assemble';
 import {defaultTieBreaker, Explicit, makeImplicit, mergeValuesWithExplicit} from '../split';
 import {UnitModel} from '../unit';
 import {LegendComponent, LegendComponentIndex} from './component';
 import * as encode from './encode';
 import * as properties from './properties';
 
+// For MVP, later remove global variable and create new types
+export let selectionOnChannel: any = [];
```

Comment:
@arvind What would be the best way to send this information to a selection transform function. We can create this list using the `model` param but `has` method of a transform does not have a `model` argument to it. In theory, we can add that argument to the function. Moreover, we can also get this list when the model is being parsed by creating a new method at `model.ts`. But, both of these methods look like a long stretch to me. I think I am missing something of a very simple solution here. Any thoughts?